sklearn/metrics/pairwise.py

```diff
@@ -527,6 +540,53 @@ def manhattan_distances(X, Y=None, sum_over_features=True,
     return D.reshape((-1, X.shape[1]))
 
 
+def _center_and_normalize(X, copy=False):
```

Comment:
That's done, thanks. I had to move the check for floats very close to machine precision into `data._handle_zeros_in_scale`, and this changes a bit the semantics of few tests for the preprocessing module. I will look into this soon.